google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -904,15 +904,15 @@ def query_job query, params: nil, external: nil,
                       legacy_sql: legacy_sql, standard_sql: standard_sql,
                       maximum_billing_tier: maximum_billing_tier,
                       maximum_bytes_billed: maximum_bytes_billed,
-                      params: params, external: external, labels: labels,
-                      udfs: udfs }
+                      job_id: job_id, prefix: prefix, params: params,
+                      external: external, labels: labels, udfs: udfs }
 
-          updater = QueryJob::Updater.from_options query, options
+          updater = QueryJob::Updater.from_options service, query, options
           updater.dataset = self
```

Comment:
Should the updater's `location` be set to `self.location` before the object is yielded?

Reply:
Yes, I'll add a commit to do this for all job methods in Dataset and Table.

Reply:
Updated.